deeplake/core/index_maintenance.py

```diff
@@ -133,7 +160,19 @@ def check_index_params(self):
     return False
 
 
+def check_index_params_text(self):
+    txt_tensor = fetch_text_tensor(self.dataset)
+    indexes = txt_tensor.get_vdb_indexes()
+    if len(indexes) == 0:
+        return False
+    else:
+        return True
+
+
 def index_operation_type_dataset(self, num_rows, changed_data_len):
+    if validate_text_tensor(self):
```

Comment:
This logic should not be here, as the `self` can object which is not a Tensor type

Reply:
Resolved.